### DISCUSSION TITLE
Img 태그에 width와 heigth를 주지 않을 경우 생기는 reflow 현상

### DISCUSSION THREAD

**Discussion by GreattitJY** [beyondDevelops/AnimalTalk#81] [General]
Img 태그 사용 시 width와 heigth의 고유 크기를 명시하지 않을 경우 브라우저 렌더링 과정에서 공간을 미리 확보하지 못하여 reflow 현상이 발생합니다.

해결 방법 1.
-  img에 width와 heigth의 크기를 둘 다 명시해야 합니다.
```html
<style>
  img {
     width: 500px;
     heigth: 300px;
  }
</style>
```
해결 방법 2.
- css에 aspect-ratio(가로,세로 비율)를 사용하고 width 혹은 heigth에 고유 크기를 명시합니다.
```html
<style>
   img {
     aspect-ratio: 4/3;
   }
</style>

<img
  src=""
  width="500"
/>
```

잘못된 예시 
- 이미지에 크기를 명시했으나 브라우저가 가로와 세로 비율을 알 수 없는 경우
```css
img {
    width: 100%, heigth: auto
}
````
위와 같은 경우 브라우저는 이미지의 가로/세로 비율을 알지 못해서 이미지가 차지하는 공간을 알지 못합니다. 따라서 이미지 다운로드가 완료된 후 다시 재렌더링을 하는 리플로우 현상이 일어납니다.

참고 문서 : https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img

**Comment by July249** [comment]
유익한 정보 공유 감사합니다! 👍

**Comment by IntHyun** [comment]
정보 공유 감사합니다!
